runners/core-java/src/main/java/org/apache/beam/runners/core/metrics/ExecutionStateTracker.java

```diff
@@ -97,6 +98,17 @@ public void onActivate(boolean pushing) {}
      */
     public abstract void reportLull(Thread trackedThread, long millis);
 
+    /**
+     * Called when a lull has been detected since the start of bundle processing. This indicates
+     * that more than {@link #BUNDLE_LULL_REPORT_MS} has been spent in the same state without any
```

Comment:
"has been spent in the same state without any transitions" appears copied from reportLull but in this case we maybe have made transitions we are just still processing the same bundle.

how about "has been spent processing the same bundle (ie between startBundle/finishBundle)."